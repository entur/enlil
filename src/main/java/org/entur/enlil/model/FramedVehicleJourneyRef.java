package org.entur.enlil.model;

import com.google.cloud.firestore.annotation.PropertyName;

public class FramedVehicleJourneyRef {

  private String dataFrameRef;

  private String datedVehicleJourneyRef;

  @PropertyName("DataFrameRef")
  public String getDataFrameRef() {
    return dataFrameRef;
  }

  @PropertyName("DataFrameRef")
  public void setDataFrameRef(String dataFrameRef) {
    this.dataFrameRef = dataFrameRef;
  }

  @PropertyName("DatedVehicleJourneyRef")
  public String getDatedVehicleJourneyRef() {
    return datedVehicleJourneyRef;
  }

  @PropertyName("DatedVehicleJourneyRef")
  public void setDatedVehicleJourneyRef(String datedVehicleJourneyRef) {
    this.datedVehicleJourneyRef = datedVehicleJourneyRef;
  }
}
