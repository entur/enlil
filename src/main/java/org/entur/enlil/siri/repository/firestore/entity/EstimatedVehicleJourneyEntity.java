package org.entur.enlil.siri.repository.firestore.entity;

import com.google.cloud.firestore.annotation.PropertyName;
import java.util.List;

public class EstimatedVehicleJourneyEntity {

  @PropertyName("EstimatedVehicleJourney")
  public EstimatedVehicleJourney EstimatedVehicleJourney;

  public static class EstimatedVehicleJourney {

    @PropertyName("RecordedAtTime")
    private String recordedAtTime;

    @PropertyName("LineRef")
    private String lineRef;

    @PropertyName("DirectionRef")
    private String directionRef;

    @PropertyName("FramedVehicleJourneyRef")
    private FramedVehicleJourneyRef framedVehicleJourneyRef;

    @PropertyName("Cancellation")
    private Boolean cancellation;

    @PropertyName("EstimatedVehicleJourneyCode")
    private String estimatedVehicleJourneyCode;

    @PropertyName("ExtraJourney")
    private Boolean extraJourney;

    @PropertyName("VehicleMode")
    private String vehicleMode;

    @PropertyName("RouteRef")
    private String routeRef;

    @PropertyName("PublishedLineName")
    private String publishedLineName;

    @PropertyName("GroupOfLinesRef")
    private String groupOfLinesRef;

    @PropertyName("ExternalLineRef")
    private String externalLineRef;

    @PropertyName("OperatorRef")
    private String operatorRef;

    @PropertyName("Monitored")
    private Boolean monitored;

    @PropertyName("DataSource")
    private String dataSource;

    @PropertyName("EstimatedCalls")
    private EstimatedCalls estimatedCalls;

    @PropertyName("IsCompleteStopSequence")
    private Boolean isCompleteStopSequence;

    @PropertyName("ExpiresAtEpochMs")
    public Long expiresAtEpochMs;

    public String getRecordedAtTime() {
      return recordedAtTime;
    }

    public void setRecordedAtTime(String recordedAtTime) {
      this.recordedAtTime = recordedAtTime;
    }

    public String getLineRef() {
      return lineRef;
    }

    public void setLineRef(String lineRef) {
      this.lineRef = lineRef;
    }

    public String getDirectionRef() {
      return directionRef;
    }

    public void setDirectionRef(String directionRef) {
      this.directionRef = directionRef;
    }

    public FramedVehicleJourneyRef getFramedVehicleJourneyRef() {
      return framedVehicleJourneyRef;
    }

    public void setFramedVehicleJourneyRef(
      FramedVehicleJourneyRef framedVehicleJourneyRef
    ) {
      this.framedVehicleJourneyRef = framedVehicleJourneyRef;
    }

    public Boolean getCancellation() {
      return cancellation;
    }

    public void setCancellation(Boolean cancellation) {
      this.cancellation = cancellation;
    }

    public String getEstimatedVehicleJourneyCode() {
      return estimatedVehicleJourneyCode;
    }

    public void setEstimatedVehicleJourneyCode(String estimatedVehicleJourneyCode) {
      this.estimatedVehicleJourneyCode = estimatedVehicleJourneyCode;
    }

    public Boolean getExtraJourney() {
      return extraJourney;
    }

    public void setExtraJourney(Boolean extraJourney) {
      this.extraJourney = extraJourney;
    }

    public String getVehicleMode() {
      return vehicleMode;
    }

    public void setVehicleMode(String vehicleMode) {
      this.vehicleMode = vehicleMode;
    }

    public String getRouteRef() {
      return routeRef;
    }

    public void setRouteRef(String routeRef) {
      this.routeRef = routeRef;
    }

    public String getPublishedLineName() {
      return publishedLineName;
    }

    public void setPublishedLineName(String publishedLineName) {
      this.publishedLineName = publishedLineName;
    }

    public String getGroupOfLinesRef() {
      return groupOfLinesRef;
    }

    public void setGroupOfLinesRef(String groupOfLinesRef) {
      this.groupOfLinesRef = groupOfLinesRef;
    }

    public String getExternalLineRef() {
      return externalLineRef;
    }

    public void setExternalLineRef(String externalLineRef) {
      this.externalLineRef = externalLineRef;
    }

    public String getOperatorRef() {
      return operatorRef;
    }

    public void setOperatorRef(String operatorRef) {
      this.operatorRef = operatorRef;
    }

    public Boolean getMonitored() {
      return monitored;
    }

    public void setMonitored(Boolean monitored) {
      this.monitored = monitored;
    }

    public String getDataSource() {
      return dataSource;
    }

    public void setDataSource(String dataSource) {
      this.dataSource = dataSource;
    }

    public EstimatedCalls getEstimatedCalls() {
      return estimatedCalls;
    }

    public void setEstimatedCalls(EstimatedCalls estimatedCalls) {
      this.estimatedCalls = estimatedCalls;
    }

    public Boolean getIsCompleteStopSequence() {
      return isCompleteStopSequence;
    }

    public void setIsCompleteStopSequence(Boolean isCompleteStopSequence) {
      isCompleteStopSequence = isCompleteStopSequence;
    }
  }

  public static class EstimatedCalls {

    @PropertyName("EstimatedCall")
    List<EstimatedCall> estimatedCall;

    public List<EstimatedCall> getEstimatedCall() {
      return estimatedCall;
    }

    public void setEstimatedCall(List<EstimatedCall> estimatedCall) {
      this.estimatedCall = estimatedCall;
    }
  }

  public static class EstimatedCall {

    @PropertyName("StopPointRef")
    private String stopPointRef;

    @PropertyName("Order")
    private Integer order;

    @PropertyName("StopPointName")
    private String stopPointName;

    @PropertyName("Cancellation")
    private Boolean cancellation;

    @PropertyName("RequestStop")
    private Boolean requestStop;

    @PropertyName("DestinationDisplay")
    private String destinationDisplay;

    @PropertyName("AimedArrivalTime")
    private String aimedArrivalTime;

    @PropertyName("ExpectedArrivalTime")
    private String expectedArrivalTime;

    @PropertyName("AimedDepartureTime")
    private String aimedDepartureTime;

    @PropertyName("ExpectedDepartureTime")
    private String expectedDepartureTime;

    @PropertyName("ArrivalStatus")
    private String arrivalStatus;

    @PropertyName("ArrivalBoardingActivity")
    private String arrivalBoardingActivity;

    @PropertyName("DepartureStatus")
    private String departureStatus;

    @PropertyName("DepartureBoardingActivity")
    private String departureBoardingActivity;

    public String getStopPointRef() {
      return stopPointRef;
    }

    public void setStopPointRef(String stopPointRef) {
      this.stopPointRef = stopPointRef;
    }

    public Integer getOrder() {
      return order;
    }

    public void setOrder(Integer order) {
      this.order = order;
    }

    public String getStopPointName() {
      return stopPointName;
    }

    public void setStopPointName(String stopPointName) {
      this.stopPointName = stopPointName;
    }

    public Boolean getCancellation() {
      return cancellation;
    }

    public void setCancellation(Boolean cancellation) {
      this.cancellation = cancellation;
    }

    public Boolean getRequestStop() {
      return requestStop;
    }

    public void setRequestStop(Boolean requestStop) {
      this.requestStop = requestStop;
    }

    public String getDestinationDisplay() {
      return destinationDisplay;
    }

    public void setDestinationDisplay(String destinationDisplay) {
      this.destinationDisplay = destinationDisplay;
    }

    public String getAimedArrivalTime() {
      return aimedArrivalTime;
    }

    public void setAimedArrivalTime(String aimedArrivalTime) {
      this.aimedArrivalTime = aimedArrivalTime;
    }

    public String getExpectedArrivalTime() {
      return expectedArrivalTime;
    }

    public void setExpectedArrivalTime(String expectedArrivalTime) {
      this.expectedArrivalTime = expectedArrivalTime;
    }

    public String getAimedDepartureTime() {
      return aimedDepartureTime;
    }

    public void setAimedDepartureTime(String aimedDepartureTime) {
      this.aimedDepartureTime = aimedDepartureTime;
    }

    public String getExpectedDepartureTime() {
      return expectedDepartureTime;
    }

    public void setExpectedDepartureTime(String expectedDepartureTime) {
      this.expectedDepartureTime = expectedDepartureTime;
    }

    public String getArrivalStatus() {
      return arrivalStatus;
    }

    public void setArrivalStatus(String arrivalStatus) {
      this.arrivalStatus = arrivalStatus;
    }

    public String getArrivalBoardingActivity() {
      return arrivalBoardingActivity;
    }

    public void setArrivalBoardingActivity(String arrivalBoardingActivity) {
      this.arrivalBoardingActivity = arrivalBoardingActivity;
    }

    public String getDepartureStatus() {
      return departureStatus;
    }

    public void setDepartureStatus(String departureStatus) {
      this.departureStatus = departureStatus;
    }

    public String getDepartureBoardingActivity() {
      return departureBoardingActivity;
    }

    public void setDepartureBoardingActivity(String departureBoardingActivity) {
      this.departureBoardingActivity = departureBoardingActivity;
    }
  }
}
