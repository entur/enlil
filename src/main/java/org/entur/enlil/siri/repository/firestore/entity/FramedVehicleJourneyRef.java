package org.entur.enlil.siri.repository.firestore.entity;

import com.google.cloud.firestore.annotation.PropertyName;

public class FramedVehicleJourneyRef {

  @PropertyName("DataFrameRef")
  private String dataFrameRef;

  @PropertyName("DatedVehicleJourneyRef")
  private String datedVehicleJourneyRef;

  public String getDataFrameRef() {
    return dataFrameRef;
  }

  public void setDataFrameRef(String dataFrameRef) {
    this.dataFrameRef = dataFrameRef;
  }

  public String getDatedVehicleJourneyRef() {
    return datedVehicleJourneyRef;
  }

  public void setDatedVehicleJourneyRef(String datedVehicleJourneyRef) {
    this.datedVehicleJourneyRef = datedVehicleJourneyRef;
  }
}
