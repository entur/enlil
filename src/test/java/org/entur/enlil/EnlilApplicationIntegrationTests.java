package org.entur.enlil;

import static java.time.format.DateTimeFormatter.ISO_DATE_TIME;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import com.google.api.gax.core.CredentialsProvider;
import com.google.api.gax.core.NoCredentialsProvider;
import com.google.cloud.firestore.Firestore;
import java.net.URI;
import java.time.Clock;
import java.time.Instant;
import java.time.OffsetDateTime;
import java.time.temporal.ChronoUnit;
import java.util.List;
import java.util.concurrent.ExecutionException;
import org.entur.enlil.siri.repository.firestore.entity.EstimatedVehicleJourneyEntity;
import org.entur.enlil.siri.repository.firestore.entity.FramedVehicleJourneyRef;
import org.entur.enlil.siri.repository.firestore.entity.PtSituationElementEntity;
import org.entur.enlil.stubs.MockedClockConfiguration;
import org.jetbrains.annotations.NotNull;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.junit.jupiter.MockitoExtension;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.boot.test.web.client.TestRestTemplate;
import org.springframework.boot.test.web.server.LocalServerPort;
import org.springframework.context.annotation.Bean;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpMethod;
import org.springframework.http.MediaType;
import org.springframework.http.RequestEntity;
import org.springframework.http.ResponseEntity;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.DynamicPropertyRegistry;
import org.springframework.test.context.DynamicPropertySource;
import org.springframework.test.context.TestPropertySource;
import org.springframework.test.context.junit.jupiter.SpringExtension;
import org.testcontainers.containers.FirestoreEmulatorContainer;
import org.testcontainers.junit.jupiter.Container;
import org.testcontainers.junit.jupiter.Testcontainers;
import org.testcontainers.utility.DockerImageName;
import uk.org.siri.siri21.Siri;

@ExtendWith(SpringExtension.class)
@SpringBootTest(
  webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
  classes = { EnlilApplication.class, MockedClockConfiguration.class }
)
@Testcontainers
@TestPropertySource("classpath:application-test.properties")
@ExtendWith({ SnapshotExtension.class, MockitoExtension.class })
@ActiveProfiles({ "test" })
class EnlilApplicationIntegrationTests {

  @LocalServerPort
  int randomPort;

  @Autowired
  Firestore firestore;

  @Autowired
  Clock clock;

  private Expect expect;

  @Container
  private static final FirestoreEmulatorContainer firestoreEmulator =
    new FirestoreEmulatorContainer(
      DockerImageName.parse("gcr.io/google.com/cloudsdktool/cloud-sdk:317.0.0-emulators")
    );

  @DynamicPropertySource
  static void emulatorProperties(DynamicPropertyRegistry registry) {
    registry.add(
      "spring.cloud.gcp.firestore.host-port",
      firestoreEmulator::getEmulatorEndpoint
    );
  }

  @TestConfiguration
  static class EmulatorConfiguration {

    // By default, autoconfiguration will initialize application default credentials.
    // For testing purposes, don't use any credentials. Bootstrap w/ NoCredentialsProvider.
    @Bean
    CredentialsProvider googleCredentials() {
      return NoCredentialsProvider.create();
    }
  }

  @Test
  void testSituationExchangeRequest() throws ExecutionException, InterruptedException {
    firestore
      .collection("codespaces/TST/authorities/TST:Authority:TST/messages")
      .add(createTestMessage())
      .get();

    TestRestTemplate restTemplate = new TestRestTemplate();
    HttpHeaders headers = new HttpHeaders();
    headers.setAccept(List.of(MediaType.APPLICATION_XML));
    headers.setContentType(MediaType.APPLICATION_XML);
    RequestEntity<String> requestEntity = new RequestEntity<>(
      """
                      <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
                      <Siri version="2.0" xmlns="http://www.siri.org.uk/siri" xmlns:ns2="http://www.ifopt.org.uk/acsb" xmlns:ns3="http://www.ifopt.org.uk/ifopt" xmlns:ns4="http://datex2.eu/schema/2_0RC1/2_0">
                        <ServiceRequest>
                          <RequestTimestamp>2019-11-06T14:45:00+01:00</RequestTimestamp>
                          <RequestorRef>ENTUR_DEV-1</RequestorRef>
                          <SituationExchangeRequest version="2.0">
                            <RequestTimestamp>2019-11-06T14:45:00+01:00</RequestTimestamp>
                            <MessageIdentifier>e11d9efb-ee7b-4a67-847a-a254e813f0da</MessageIdentifier>
                          </SituationExchangeRequest>
                        </ServiceRequest>
                      </Siri>
                      """,
      headers,
      HttpMethod.POST,
      URI.create("http://localhost:" + randomPort + "/siri"),
      Siri.class
    );

    ResponseEntity<String> response = restTemplate.exchange(requestEntity, String.class);

    expect.toMatchSnapshot(response.getBody());
  }

  PtSituationElementEntity createTestMessage() {
    var message = new PtSituationElementEntity();
    message.setCreationTime("2023-09-19T14:45:00+01:00");
    message.setParticipantRef("TST");
    message.setSituationNumber("TST:SituationNumber:1");
    PtSituationElementEntity.Source source = new PtSituationElementEntity.Source();
    source.setSourceType("directReport");
    message.setSource(source);
    message.setProgress("open");
    message.setSeverity("normal");
    message.setReportType("general");
    PtSituationElementEntity.Text summary = new PtSituationElementEntity.Text();
    summary.setText("we have a situation here");
    PtSituationElementEntity.Attributes attributes =
      new PtSituationElementEntity.Attributes();
    attributes.setXmlLang("no");
    summary.setAttributes(attributes);
    message.setSummary(summary);
    PtSituationElementEntity.Affects affects = createAffects();
    message.setAffects(affects);
    PtSituationElementEntity.ValidityPeriod validityPeriod =
      new PtSituationElementEntity.ValidityPeriod();
    validityPeriod.setStartTime("2023-09-19T14:45:00+01:00");
    message.setValidityPeriod(validityPeriod);
    return message;
  }

  private static PtSituationElementEntity.@NotNull Affects createAffects() {
    PtSituationElementEntity.Affects affects = new PtSituationElementEntity.Affects();
    PtSituationElementEntity.Networks networks = new PtSituationElementEntity.Networks();
    PtSituationElementEntity.AffectedNetwork affectedNetwork =
      new PtSituationElementEntity.AffectedNetwork();
    PtSituationElementEntity.AffectedLine affectedLine =
      new PtSituationElementEntity.AffectedLine();
    affectedLine.setLineRef("TST:Line:1");
    affectedNetwork.setAffectedLine(affectedLine);
    networks.setAffectedNetwork(affectedNetwork);
    affects.setNetworks(networks);
    return affects;
  }

  @Test
  void testEstimatedTimetableRequestWithCancellation()
    throws ExecutionException, InterruptedException {
    firestore
      .collection("codespaces/TST/authorities/TST:Authority:TST/cancellations")
      .add(createTestCancellation())
      .get();

    TestRestTemplate restTemplate = new TestRestTemplate();

    HttpHeaders headers = new HttpHeaders();
    headers.setAccept(List.of(MediaType.APPLICATION_XML));
    headers.setContentType(MediaType.APPLICATION_XML);
    RequestEntity<String> requestEntity = new RequestEntity<>(
      """
                    <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
                    <Siri version="2.0" xmlns="http://www.siri.org.uk/siri" xmlns:ns2="http://www.ifopt.org.uk/acsb" xmlns:ns3="http://www.ifopt.org.uk/ifopt" xmlns:ns4="http://datex2.eu/schema/2_0RC1/2_0">
                      <ServiceRequest>
                        <RequestTimestamp>2019-11-06T14:45:00+01:00</RequestTimestamp>
                        <RequestorRef>ENTUR_DEV-1</RequestorRef>
                        <EstimatedTimetableRequest version="2.0">
                          <RequestTimestamp>2019-11-06T14:45:00+01:00</RequestTimestamp>
                          <MessageIdentifier>e11d9efb-ee7b-4a67-847a-a254e813f0da</MessageIdentifier>
                        </EstimatedTimetableRequest>
                      </ServiceRequest>
                    </Siri>
                    """,
      headers,
      HttpMethod.POST,
      URI.create("http://localhost:" + randomPort + "/siri"),
      Siri.class
    );

    ResponseEntity<String> response = restTemplate.exchange(requestEntity, String.class);

    expect.toMatchSnapshot(response.getBody());
  }

  private EstimatedVehicleJourneyEntity createTestCancellation() {
    var cancellation = new EstimatedVehicleJourneyEntity();
    var estimatedVehicleJourney =
      new EstimatedVehicleJourneyEntity.EstimatedVehicleJourney();
    estimatedVehicleJourney.setRecordedAtTime(
      ISO_DATE_TIME.format(OffsetDateTime.now(clock))
    );
    estimatedVehicleJourney.setLineRef("TST:Line:1");
    estimatedVehicleJourney.setDirectionRef("0");
    FramedVehicleJourneyRef framedVehicleJourneyRef = new FramedVehicleJourneyRef();
    framedVehicleJourneyRef.setDataFrameRef("TST:ServiceJourney:1");
    framedVehicleJourneyRef.setDatedVehicleJourneyRef("2024-01-01");
    estimatedVehicleJourney.setFramedVehicleJourneyRef(framedVehicleJourneyRef);
    estimatedVehicleJourney.setCancellation(true);
    estimatedVehicleJourney.setDataSource("TST");
    estimatedVehicleJourney.setEstimatedCalls(createTestEstimatedCalls());
    estimatedVehicleJourney.setIsCompleteStopSequence(true);
    estimatedVehicleJourney.setExpiresAtEpochMs(
      Instant.now(clock).plus(20, ChronoUnit.MINUTES).toEpochMilli()
    );

    cancellation.setEstimatedVehicleJourney(estimatedVehicleJourney);

    return cancellation;
  }

  private EstimatedVehicleJourneyEntity.@NotNull EstimatedCalls createTestEstimatedCalls() {
    EstimatedVehicleJourneyEntity.EstimatedCalls estimatedCalls =
      new EstimatedVehicleJourneyEntity.EstimatedCalls();
    EstimatedVehicleJourneyEntity.EstimatedCall call1 =
      new EstimatedVehicleJourneyEntity.EstimatedCall();
    call1.setStopPointRef("TST:Quay:1");
    call1.setOrder(1);
    call1.setStopPointName("Fjord");
    call1.setCancellation(true);
    call1.setRequestStop(false);
    call1.setArrivalBoardingActivity("noAlighting");
    call1.setDepartureStatus("cancelled");
    call1.setDepartureBoardingActivity("boarding");
    EstimatedVehicleJourneyEntity.EstimatedCall call2 =
      new EstimatedVehicleJourneyEntity.EstimatedCall();
    call2.setStopPointRef("TST:Quay:2");
    call2.setOrder(2);
    call2.setStopPointName("Fjell");
    call2.setCancellation(true);
    call2.setRequestStop(false);
    call2.setArrivalBoardingActivity("noAlighting");
    call2.setDepartureStatus("cancelled");
    call2.setDepartureBoardingActivity("boarding");
    estimatedCalls.setEstimatedCall(List.of(call1, call2));
    return estimatedCalls;
  }

  @Test
  void testEstimatedTimetableRequestWithExtraJourney()
    throws ExecutionException, InterruptedException {
    firestore
      .collection("codespaces/TST/authorities/TST:Authority:TST/extrajourneys")
      .add(createTestExtraJourney())
      .get();

    TestRestTemplate restTemplate = new TestRestTemplate();

    HttpHeaders headers = new HttpHeaders();
    headers.setAccept(List.of(MediaType.APPLICATION_XML));
    headers.setContentType(MediaType.APPLICATION_XML);
    RequestEntity<String> requestEntity = new RequestEntity<>(
      """
                          <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
                          <Siri version="2.0" xmlns="http://www.siri.org.uk/siri" xmlns:ns2="http://www.ifopt.org.uk/acsb" xmlns:ns3="http://www.ifopt.org.uk/ifopt" xmlns:ns4="http://datex2.eu/schema/2_0RC1/2_0">
                            <ServiceRequest>
                              <RequestTimestamp>2019-11-06T14:45:00+01:00</RequestTimestamp>
                              <RequestorRef>ENTUR_DEV-1</RequestorRef>
                              <EstimatedTimetableRequest version="2.0">
                                <RequestTimestamp>2019-11-06T14:45:00+01:00</RequestTimestamp>
                                <MessageIdentifier>e11d9efb-ee7b-4a67-847a-a254e813f0da</MessageIdentifier>
                              </EstimatedTimetableRequest>
                            </ServiceRequest>
                          </Siri>
                          """,
      headers,
      HttpMethod.POST,
      URI.create("http://localhost:" + randomPort + "/siri"),
      Siri.class
    );

    ResponseEntity<String> response = restTemplate.exchange(requestEntity, String.class);

    expect.toMatchSnapshot(response.getBody());
  }

  private EstimatedVehicleJourneyEntity createTestExtraJourney() {
    var extraJourney = new EstimatedVehicleJourneyEntity();
    var estimatedVehicleJourney =
      new EstimatedVehicleJourneyEntity.EstimatedVehicleJourney();
    estimatedVehicleJourney.setRecordedAtTime(
      ISO_DATE_TIME.format(OffsetDateTime.now(clock))
    );
    estimatedVehicleJourney.setLineRef("TST:Line:1");
    estimatedVehicleJourney.setDirectionRef("0");
    estimatedVehicleJourney.setDataSource("TST");
    estimatedVehicleJourney.setEstimatedCalls(createExtraJourneyTestEstimatedCalls());
    estimatedVehicleJourney.setIsCompleteStopSequence(true);
    estimatedVehicleJourney.setExpiresAtEpochMs(
      Instant.now(clock).plus(20, ChronoUnit.MINUTES).toEpochMilli()
    );

    estimatedVehicleJourney.setExternalLineRef("TST:Line:1");
    estimatedVehicleJourney.setGroupOfLinesRef("TST:Network:1");
    estimatedVehicleJourney.setOperatorRef("TST:Operator:1");
    estimatedVehicleJourney.setPublishedLineName("Test");
    estimatedVehicleJourney.setRouteRef("TST:Route:1");
    estimatedVehicleJourney.setVehicleMode("bus");

    extraJourney.setEstimatedVehicleJourney(estimatedVehicleJourney);
    return extraJourney;
  }

  private EstimatedVehicleJourneyEntity.EstimatedCalls createExtraJourneyTestEstimatedCalls() {
    EstimatedVehicleJourneyEntity.EstimatedCalls estimatedCalls =
      new EstimatedVehicleJourneyEntity.EstimatedCalls();
    EstimatedVehicleJourneyEntity.EstimatedCall call1 =
      new EstimatedVehicleJourneyEntity.EstimatedCall();
    call1.setStopPointRef("TST:Quay:1");
    call1.setOrder(1);
    call1.setStopPointName("Fjord");
    call1.setDepartureBoardingActivity("boarding");
    EstimatedVehicleJourneyEntity.EstimatedCall call2 =
      new EstimatedVehicleJourneyEntity.EstimatedCall();
    call2.setStopPointRef("TST:Quay:2");
    call2.setOrder(2);
    call2.setStopPointName("Fjell");
    call2.setArrivalBoardingActivity("alighting");
    estimatedCalls.setEstimatedCall(List.of(call1, call2));
    return estimatedCalls;
  }
}
