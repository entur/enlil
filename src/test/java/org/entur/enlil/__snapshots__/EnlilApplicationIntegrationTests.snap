org.entur.enlil.EnlilApplicationIntegrationTests.testEstimatedTimetableRequestWithCancellation=[
<?xml version="1.0" encoding="UTF-8" standalone="yes"?><Siri xmlns="http://www.siri.org.uk/siri" xmlns:ns2="http://www.ifopt.org.uk/acsb" xmlns:ns3="http://www.ifopt.org.uk/ifopt" xmlns:ns4="http://datex2.eu/schema/2_0RC1/2_0" xmlns:ns5="http://www.opengis.net/gml/3.2" version="2.1"><ServiceDelivery><ResponseTimestamp>2023-09-19T19:19:00Z</ResponseTimestamp><ProducerRef>ENT</ProducerRef><EstimatedTimetableDelivery version="2.1"><ResponseTimestamp>2023-09-19T19:19:00Z</ResponseTimestamp><EstimatedJourneyVersionFrame><RecordedAtTime>2023-09-19T19:19:00Z</RecordedAtTime><EstimatedVehicleJourney><RecordedAtTime>2023-09-19T19:19:00Z</RecordedAtTime><LineRef>TST:Line:1</LineRef><DirectionRef>0</DirectionRef><FramedVehicleJourneyRef><DataFrameRef>TST:ServiceJourney:1</DataFrameRef><DatedVehicleJourneyRef>2024-01-01</DatedVehicleJourneyRef></FramedVehicleJourneyRef><Cancellation>true</Cancellation><DataSource>TST</DataSource><EstimatedCalls><EstimatedCall><StopPointRef>TST:Quay:1</StopPointRef><Order>1</Order><StopPointName>Fjord</StopPointName><Cancellation>true</Cancellation><RequestStop>false</RequestStop><ArrivalBoardingActivity>noAlighting</ArrivalBoardingActivity><DepartureStatus>cancelled</DepartureStatus><DepartureBoardingActivity>boarding</DepartureBoardingActivity></EstimatedCall><EstimatedCall><StopPointRef>TST:Quay:2</StopPointRef><Order>2</Order><StopPointName>Fjell</StopPointName><Cancellation>true</Cancellation><RequestStop>false</RequestStop><ArrivalBoardingActivity>noAlighting</ArrivalBoardingActivity><DepartureStatus>cancelled</DepartureStatus><DepartureBoardingActivity>boarding</DepartureBoardingActivity></EstimatedCall></EstimatedCalls><IsCompleteStopSequence>true</IsCompleteStopSequence></EstimatedVehicleJourney></EstimatedJourneyVersionFrame></EstimatedTimetableDelivery></ServiceDelivery></Siri>
]


org.entur.enlil.EnlilApplicationIntegrationTests.testEstimatedTimetableRequestWithExtraJourney=[
<?xml version="1.0" encoding="UTF-8" standalone="yes"?><Siri xmlns="http://www.siri.org.uk/siri" xmlns:ns2="http://www.ifopt.org.uk/acsb" xmlns:ns3="http://www.ifopt.org.uk/ifopt" xmlns:ns4="http://datex2.eu/schema/2_0RC1/2_0" xmlns:ns5="http://www.opengis.net/gml/3.2" version="2.1"><ServiceDelivery><ResponseTimestamp>2023-09-19T19:19:00Z</ResponseTimestamp><ProducerRef>ENT</ProducerRef><EstimatedTimetableDelivery version="2.1"><ResponseTimestamp>2023-09-19T19:19:00Z</ResponseTimestamp><EstimatedJourneyVersionFrame><RecordedAtTime>2023-09-19T19:19:00Z</RecordedAtTime><EstimatedVehicleJourney><RecordedAtTime>2023-09-19T19:19:00Z</RecordedAtTime><LineRef>TST:Line:1</LineRef><DirectionRef>0</DirectionRef><FramedVehicleJourneyRef><DataFrameRef>TST:ServiceJourney:1</DataFrameRef><DatedVehicleJourneyRef>2024-01-01</DatedVehicleJourneyRef></FramedVehicleJourneyRef><Cancellation>true</Cancellation><DataSource>TST</DataSource><EstimatedCalls><EstimatedCall><StopPointRef>TST:Quay:1</StopPointRef><Order>1</Order><StopPointName>Fjord</StopPointName><Cancellation>true</Cancellation><RequestStop>false</RequestStop><ArrivalBoardingActivity>noAlighting</ArrivalBoardingActivity><DepartureStatus>cancelled</DepartureStatus><DepartureBoardingActivity>boarding</DepartureBoardingActivity></EstimatedCall><EstimatedCall><StopPointRef>TST:Quay:2</StopPointRef><Order>2</Order><StopPointName>Fjell</StopPointName><Cancellation>true</Cancellation><RequestStop>false</RequestStop><ArrivalBoardingActivity>noAlighting</ArrivalBoardingActivity><DepartureStatus>cancelled</DepartureStatus><DepartureBoardingActivity>boarding</DepartureBoardingActivity></EstimatedCall></EstimatedCalls><IsCompleteStopSequence>true</IsCompleteStopSequence></EstimatedVehicleJourney><EstimatedVehicleJourney><RecordedAtTime>2023-09-19T19:19:00Z</RecordedAtTime><LineRef>TST:Line:1</LineRef><DirectionRef>0</DirectionRef><VehicleMode>bus</VehicleMode><RouteRef>TST:Route:1</RouteRef><PublishedLineName>Test</PublishedLineName><GroupOfLinesRef>TST:Network:1</GroupOfLinesRef><ExternalLineRef>TST:Line:1</ExternalLineRef><OperatorRef>TST:Operator:1</OperatorRef><DataSource>TST</DataSource><EstimatedCalls><EstimatedCall><StopPointRef>TST:Quay:1</StopPointRef><Order>1</Order><StopPointName>Fjord</StopPointName><DepartureBoardingActivity>boarding</DepartureBoardingActivity></EstimatedCall><EstimatedCall><StopPointRef>TST:Quay:2</StopPointRef><Order>2</Order><StopPointName>Fjell</StopPointName><ArrivalBoardingActivity>alighting</ArrivalBoardingActivity></EstimatedCall></EstimatedCalls><IsCompleteStopSequence>true</IsCompleteStopSequence></EstimatedVehicleJourney></EstimatedJourneyVersionFrame></EstimatedTimetableDelivery></ServiceDelivery></Siri>
]


org.entur.enlil.EnlilApplicationIntegrationTests.testSituationExchangeRequest=[
<?xml version="1.0" encoding="UTF-8" standalone="yes"?><Siri xmlns="http://www.siri.org.uk/siri" xmlns:ns2="http://www.ifopt.org.uk/acsb" xmlns:ns3="http://www.ifopt.org.uk/ifopt" xmlns:ns4="http://datex2.eu/schema/2_0RC1/2_0" xmlns:ns5="http://www.opengis.net/gml/3.2" version="2.1"><ServiceDelivery><ResponseTimestamp>2023-09-19T19:19:00Z</ResponseTimestamp><ProducerRef>ENT</ProducerRef><SituationExchangeDelivery><ResponseTimestamp>2023-09-19T19:19:00Z</ResponseTimestamp><Situations><PtSituationElement><CreationTime>2023-09-19T13:45:00Z</CreationTime><ParticipantRef>TST</ParticipantRef><SituationNumber>TST:SituationNumber:1</SituationNumber><Source><SourceType>directReport</SourceType></Source><Progress>open</Progress><ValidityPeriod><StartTime>2023-09-19T13:45:00Z</StartTime></ValidityPeriod><UndefinedReason></UndefinedReason><Severity>normal</Severity><ReportType>general</ReportType><Summary xml:lang="no">we have a situation here</Summary><Affects><Networks><AffectedNetwork><AffectedLine><LineRef>TST:Line:1</LineRef></AffectedLine></AffectedNetwork></Networks></Affects></PtSituationElement></Situations></SituationExchangeDelivery></ServiceDelivery></Siri>
]